### PR TITLE
Switch to CircleCI

### PR DESCRIPTION
**Pull request by SuperQ**
Replace Travis with CircleCI for running tests.

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

**Review by reimda — COMMENTED** [review]
Looks good to me

**Comment by brian-brazil** [comment]
:+1:

**Review by clarkmcc — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    - run: sudo /etc/init.d/snmpd restart
+    - run: make tools
+    - run: make lint
+    - run: go test -v -tags helper
+    - run: go test -v -tags marshal
+    - run: go test -v -tags misc
+    - run: go test -v -tags api
+    - run: go test -v -tags end2end
+    - run: go test -v -tags trap
+
```
Comment (TimRots) [inline .circleci/config.yml:38]: Could you also add `` -run: go test -v -tags all -race``

Comment (SuperQ) [reply .circleci/config.yml:39]:
```
go test: -race is only supported on linux/amd64, linux/ppc64le, linux/arm64, freebsd/amd64, netbsd/amd64, darwin/amd64 and windows/amd64
```

Whups, doesn't worth with the 386 builds.

**Comment by TimRots** [comment]
A nice to have would be to integrate [staticcheck](https://staticcheck.io) to ensure no known bad-practices slip in and reduce review work on future PR's.

**Comment by SuperQ** [comment]
@TimRots Yes, let's add that in a future PR.

**Review by TimRots — APPROVED** [review]

**Pull request merged**